src/test/python/parser/opentrackio_lib.py

```diff
@@ -8,67 +8,175 @@
 import json
 import os
 import math
+import struct
+from datetime import datetime, timedelta
+from enum import Enum
+from cbor2 import dumps, loads, load
+from jsonschema import validate, ValidationError, SchemaError
+from typing import Optional
+
+OTRK_VERSION = "0.9.2"
+OTRK_IDENTIFIER = b'OTrk' 
+OTRK_IDENTIFIER_LENGTH = 4
+OTRK_HEADER_LENGTH = 16
+OTRK_SOURCE_NUMBER = 1
+OTRK_MULTICAST_PREFIX =  f"235.135.1."
+OTRK_MULTICAST_PORT = 55555
+OTRK_MTU = 1500  # Maximum Transmission Unit (bytes)
+OTRK_MAX_PAYLOAD_SIZE = OTRK_MTU - OTRK_HEADER_LENGTH
+
+NTPSERVER = "pool.ntp.org"
+
+class Translation(Enum):
+    X = "x"
+    Y = "y"
+    Z = "z"
+    
+class Rotation(Enum):
+    PAN = "pan"
+    TILT = "tilt"
+    ROLL = "roll"
+    
+class TranslationUnit(Enum):
+    METER = "m"
+    CENTIMETER = "cm"
+    MILLIMETER = "mm"
+    INCH = "in"
+    
+    def conversion_factor_from_meters(self) -> float:
+        conversion_factors = {
+            "m": 1.0,
+            "cm": 100.0,
+            "mm": 1000.0,
+            "in": 1000.0 / 25.4,
+        }
+        return conversion_factors[self.value]
+        
+class RotationUnit(Enum):
+    DEGREE = "deg"
+    RADIAN = "rad"
+    
+    def conversion_factor_from_degrees(self) -> float:
+        conversion_factors = {
+            "deg": 1.0,
+            "rad": math.pi/180
+        }
+        return conversion_factors[self.value]
+
+class TimeFormat(Enum):
+    SECONDS = "sec"
+    TIMECODE = "tc"
+    STRING = "string"
+    ISO8601 = "iso8601"
+    
+class FocusDistanceUnit(Enum):
+    METER = "m"
+    CENTIMETER = "cm"
+    MILLIMETER = "mm"
+    INCH = "in"
+    
+    def conversion_factor_from_mm(self) -> float:
+        conversion_factors = {
+            "m": 0.001,
+            "cm": 0.1,
+            "mm": 1.0,
+            "in": 1 / 25.4
+        }    
+        return conversion_factors[self.value]
+        
+class TimeSource(Enum):
+    PTP = "ptp"
+    NTP = "ntp"
+    GENLOCK = "genlock"
+    NONE = "none"
+    
+class PayloadFormat(Enum):
+    JSON = 0x01
+    CBOR = 0x02
+
+def fletcher16(data: bytes) -> bytes:
```

Comment:
Yes its good to keep this implementation in our code since we want the checkum to succeed and end-users may have trouble finding this alg on the WWW.